src/VisualStudio/LiveShare/Impl/Client/RemoteLanguageServiceWorkspace.cs

```diff
@@ -311,19 +365,23 @@ public void NotifyOnDocumentClosing(string moniker)
         /// <inheritdoc />
         public override void OpenDocument(DocumentId documentId, bool activate = true)
         {
+            if (!IsRemoteSession)
```

Comment:
```suggestion
            if (_session == null)
```

If you write this, does the ! on _session later become unnecessary?